src/MoBi.Presentation/Tasks/Interaction/InteractionTasksForContainerBase.cs

```diff
@@ -38,9 +37,9 @@ public override IContainer CreateNewEntity(TParent parent)
          return newEntity;
       }
 
-      protected MoBiSpatialStructure GetSpatialStructure()
+      private MoBiSpatialStructure getSpatialStructure(IBuildingBlock buildingBlockWithFormulaCache)
       {
-         return _interactionTaskContext.Active<MoBiSpatialStructure>();
+         return buildingBlockWithFormulaCache as MoBiSpatialStructure ?? _interactionTaskContext.Active<MoBiSpatialStructure>();
```

Comment:
pretty sure this is the correct way of doing this since the building block is passed as a parameter..but...who knows